app/mailers/application_mailer.rb

```diff
@@ -17,6 +17,16 @@ def default_from_address
     ENV[ 'MAILER_SENDER' ]
   end
 
+  def ahoy_user( email_address, name = nil )
+    @user = User.find_by( email: email_address )
+    return if @user.present?
+
+    @user = EmailRecipient.find_by( email: email_address )
+    return if @user.present?
+
+    @user = EmailRecipient.create!( name: name, email: email_address )
+  end
```

Comment:
The diff below doesn't do much for me, but its a good opportunity to show pairing activerecord's `create_with` and `find_or_create_by(!)` methods: https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by

```diff
-    @user = EmailRecipient.find_by( email: email_address )
-    return if @user.present?

-    @user = EmailRecipient.create!( name: name, email: email_address )
+    @user = EmailRecipient.create_with(name: name).find_or_create_by!(email: email_address)
```